### PR TITLE
Some fix

### PR DESCRIPTION
**Pull request by yuehhua**

### CODE REVIEW

File: benchmark/FieldDistributionNonuniformMedium/src/FieldDistributionNonuniformMedium.jl [hunk 1]
```diff
@@ -5,86 +5,71 @@ using Plots
...
 const C = 299792458
+const MAX_X = 3e-6
+const MAX_Y = 10e-6
+const MAX_T = 5e-12
 
-struct Bound
-    max_x::Float64
-    max_y::Float64
-    max_t::Float64
-end
```
Comment (foldfelis) [inline benchmark/FieldDistributionNonuniformMedium/src/FieldDistributionNonuniformMedium.jl]: Those are tunable params

File: benchmark/FieldDistributionNonuniformMedium/src/FieldDistributionNonuniformMedium.jl [hunk 2]
```diff
@@ -93,33 +78,32 @@ mutable struct Simulator
     hx::Matrix{Float64}
     hy::Matrix{Float64}
 
-    t::Int64
+    nx::Int
+    ny::Int
+    nt::Int
+    Δx
+    Δt
+    t::Int
```
Comment (foldfelis) [inline benchmark/FieldDistributionNonuniformMedium/src/FieldDistributionNonuniformMedium.jl]: I'm not quite sure if this is a good idea to decapsulate `Discretizer` like this...

**Comment by yuehhua** [comment]
About variables, there are some practices:

1. Make it a constant. This practice will make variables fix and hard to tune with.
2. Make it a function argument. This practice will make it really flexible and dynamic.
3. Make it an object field. This practice will make an object stateful and memorize the variable.

Basically, the design depends on the context that user use the objects. Maybe you can list your user story and we can discuss.

**Comment by yuehhua** [comment]
@foldfelis You can think about it, and I will get a new version for you.

**Comment by yuehhua** [comment]
I add `GridParameter` and it absorbes functionality of `Bound` and `Discretizer`.
`GridParameter` can be generated by `discretize` from some parameters.
`gen_grid` is merged as a part of method, that is `init`, for `GridParameter`.
And some array-like methods (e.g. `size`, `axes`) are added for `GridParameter` for more Julia-like syntax.
`boundary` can access the maximum of x and y from `GridParameter`.
All of them, dimensions can be specified.

**Comment by yuehhua** [comment]
I guess @foldfelis want to decouple `GridParameter` from `Simulator`, right? That way, these two objects can be independent.

**Comment by foldfelis** [comment]
Looks good to me, BTW, is there a better function name for `init(gp)` because this function is more like a grid generator.

File: benchmark/FieldDistributionNonuniformMedium/src/FieldDistributionNonuniformMedium.jl [hunk 3]
```diff
@@ -2,89 +2,103 @@ module FieldDistributionNonuniformMedium
...
+    Δx = max_x / nx
+    Δy = max_y / ny
 
     Δt = 1 / C / √(1/Δx^2 + 1/Δy^2)
-    nt = round(Int, b.max_t/Δt)
+    nt = round(Int, max_t/Δt)
 
-    return Discretizer(nx, ny, Δx, Δy, Δt, nt)
+    return GridParameter(nx, ny, nt, Δx, Δy, Δt, max_x, max_y, max_t)
 end
```
Comment (foldfelis) [inline benchmark/FieldDistributionNonuniformMedium/src/FieldDistributionNonuniformMedium.jl:32]: This function looks like a cconstructor?

**Comment by yuehhua** [comment]
Yeah, there should be a better name for `init(gp)`. I just have no idea how to name it.
It can be `generate`, `build` or `create`.

**Pull request merged**